File: Implementation/boj_5525.py
n = int(input())
m = int(input())
s = input()

cnt = 0
i = 0

while i < (m - 1):
    if s[i:i + 3] == "IOI":
        k = 0
        while i + 2 < m and s[i:i + 3] == "IOI":
            k += 1
            i += 2
        if k >= n:
            cnt += k - n + 1
    else:
        i += 1

print(cnt)

File: Implementation/boj_5525.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    int n, m;
    cin >> n >> m;
    string s;
    cin >> s;

    int cnt = 0;
    int i = 0;

    while (i < m - 1) {
        if (s.substr(i, 3) == "IOI") {
            int k = 0;
            while (i + 2 < m && s.substr(i, 3) == "IOI") {
                k++;
                i += 2;
            }
            if (k >= n) {
                cnt += k - n + 1;
            }
        } else {
            i++;
        }
    }

    cout << cnt << endl;

    return 0;
}
